README.md
# ritik.ml
Link to my personal website: https://ritik.ml/

## Stack
- Built on `Jekyll`.
- `Lanyon` theme.
- Hosted on [Github pages](https://pages.github.com/)

## Inspiration
Parts of this project are 'inspired' from these sources:
- https://github.com/captn3m0/captn3m0.github.com
- https://github.com/jogendra/jogendra.github.io

## License
`Lanyon` Jekyll theme is under the [MIT license](https://github.com/poole/lanyon/blob/master/LICENSE.md).


projects/news.md
---
repo: dev-ritik/news
layout: repo
permalink: /news/
title: news
---


mainpage.js
$(document).ready(function(){
	$(".icon").hover(function(){
		var currentId=this.id;
		 var $reqs = $("."+currentId+"");
		 var $el = $reqs.eq(0); 
		 var $el1 = $reqs.eq(1); 
		$el.toggleClass('icon_hover');
		$el1.toggleClass('name_hover');
	});

	$(".name").hover(function(){
		var currentId=this.id;
		 var $reqs = $("."+currentId+"");
		 var $el = $reqs.eq(0); 
		 var $el1 = $reqs.eq(1); 
		$el.toggleClass('icon_hover');
		$el1.toggleClass('name_hover');
	});
});

var TxtRotate = function(el, toRotate, period) {
  this.toRotate = toRotate;
  this.el = el;
  this.loopNum = 0;
  this.period = parseInt(period, 10) || 2000;
  this.txt = '';
  this.tick();
  this.isDeleting = false;
};

TxtRotate.prototype.tick = function() {
  var i = this.loopNum % this.toRotate.length;
  var fullTxt = this.toRotate[i];

  if (this.isDeleting) {
    this.txt = fullTxt.substring(0, this.txt.length - 1);
  } else {
    this.txt = fullTxt.substring(0, this.txt.length + 1);
  }

  this.el.innerHTML = '<span class="wrap">'+this.txt+'</span>';

  var that = this;
  var delta = 150 - Math.random() * 100;

  if (this.isDeleting) { delta /= 2; }

  if (!this.isDeleting && this.txt === fullTxt) {
    delta = this.period;
    this.isDeleting = true;
  } else if (this.isDeleting && this.txt === '') {
    this.isDeleting = false;
    this.loopNum++;
    delta = 500;
  }

  setTimeout(function() {
    that.tick();
  }, delta);
};

window.onload = function() {
  var elements = document.getElementsByClassName('txt-rotate');
  for (var i=0; i<elements.length; i++) {
    var toRotate = elements[i].getAttribute('data-rotate');
    var period = elements[i].getAttribute('data-period');
    if (toRotate) {
      new TxtRotate(elements[i], JSON.parse(toRotate), period);
    }
  }
};



src/mainpage.ts
$(function (): void {
  $('.icon').on('mouseenter mouseleave', function (): void {
    const currentId = this.id;
    const $reqs = $('.' + currentId + '');
    const $el = $reqs.eq(0);
    const $el1 = $reqs.eq(1);
    $el.toggleClass('icon_hover');
    $el1.toggleClass('name_hover');
  });

  $('.name').on('mouseenter mouseleave', function (): void {
    const currentId = this.id;
    const $reqs = $('.' + currentId + '');
    const $el = $reqs.eq(0);
    const $el1 = $reqs.eq(1);
    $el.toggleClass('icon_hover');
    $el1.toggleClass('name_hover');
  });
});

class TxtRotate {
  private readonly toRotate: string[];
  private el: any;
  private loopNum: number;
  private readonly period: number;
  private txt: string;
  private isDeleting: boolean;

  constructor(el: HTMLElement, toRotate: string[], period: string) {
    this.toRotate = toRotate;
    this.el = el;
    this.loopNum = 0;
    this.period = parseInt(period, 10) || 2000;
    this.txt = '';
  }

  public start(): void {
    this.tick();
    this.isDeleting = false;
  }

  private tick(): void {
    const i = this.loopNum % this.toRotate.length;
    const fullTxt = this.toRotate[i];

    if (this.isDeleting) {
      this.txt = fullTxt.substring(0, this.txt.length - 1);
    } else {
      this.txt = fullTxt.substring(0, this.txt.length + 1);
    }

    this.el.innerHTML = '<span class="wrap">' + this.txt + '</span>';

    let delta = 150 - Math.random() * 100;

    if (this.isDeleting) {
      delta /= 2;
    }

    if (!this.isDeleting && this.txt === fullTxt) {
      delta = this.period;
      this.isDeleting = true;
    } else if (this.isDeleting && this.txt === '') {
      this.isDeleting = false;
      this.loopNum++;
      delta = 500;
    }

    setTimeout(() => {
      this.tick();
    }, delta);
  }
}

window.onload = function (): void {
  const elements: HTMLCollectionOf<any> = document.getElementsByClassName('txt-rotate');
  // @ts-ignore
  for (const element of elements) {
    const toRotate = element.getAttribute('data-rotate');
    const period = element.getAttribute('data-period');
    if (toRotate) {
      const rotatable: TxtRotate = new TxtRotate(element, JSON.parse(toRotate), period);
      rotatable.start();
    }
  }
};


webpack.common.js
const path = require("path");

module.exports = {
    entry: {
        mainpage: path.join(__dirname, "src/mainpage.ts"),
        background: path.join(__dirname, "src/background.ts")
    },
    output: {
        path: path.join(__dirname, "dist/js/"),
        filename: "[name].js"
    },
    module: {
        rules: [{
                exclude: /node_modules/,
                test: /\.tsx?$/,
                use: "ts-loader"
            },
            {
                exclude: /node_modules/,
                test: /\.scss$/,
                use: [{
                        loader: "style-loader" // Creates style nodes from JS strings
                    },
                    {
                        loader: "css-loader" // Translates CSS into CommonJS
                    },
                    {
                        loader: "sass-loader" // Compiles Sass to CSS
                    }
                ]
            }
        ]
    },
    resolve: {
        extensions: [".ts", ".tsx", ".js"]
    }
};

background/background.js
var context;
var array = [];
var update;
var reset;
var canvas;
//update timer every second//update timer every second

function startAnim() {
  for(var i=0;i<5;i++){
    array.push(new component( generateRandom(0.5,4), "white", window.innerWidth/2, window.innerHeight/2));
  }
}

(function(){
  console.log("here");
  window.onfocus=function(){
    console.log("here1"+reset+update);
    if(!reset)
      reset=window.setInterval(startAnim,280);
    if(!update)
      update=window.setInterval(updateCanvas,20);
    console.log("here1"+reset+update);
    };

    window.onblur=function(){
    console.log("here2"+reset+update);
    if(reset){
      window.clearInterval(reset);
      reset=null;
    }
    if(update){
      window.clearInterval(update);
      update=null;
    }
    console.log("here2"+reset+update);

  };

  window.onresize = function(event) {
    refreshCanvas();
  }
})();

$(document).ready(function(){
  window.focus();

});
function refreshCanvas(){
  canvas.width=window.innerWidth;
  canvas.height=window.innerHeight;
}

function makeCanvas(){
  startAnim();
  canvas = document.getElementById('canvas');
  refreshCanvas();
    if (canvas.getContext) {
      context = this.canvas.getContext("2d");
      document.body.insertBefore(this.canvas, document.body.childNodes[0]);
    }

  return context;
}

class component {
  constructor(radius, color, x, y) {
    this.radius = radius;
    this.x = x;
    this.y = y;
    this.setSpeed();
    this.color=color;
    this.rate=Math.random()/150+0.02;
  }

  setSpeed = function () {
    this.speedX = (generateRandom(-3,3)); //represents movements in x direction per game update
    this.speedY = (generateRandom(-3,3));
    if (Math.abs(this.speedX) + Math.abs(this.speedY) < 2) {
      this.setSpeed();
    }
  }

  update = function(ctx) {
    ctx.beginPath();
    ctx.arc(this.x,this.y,this.radius,0,2*Math.PI);
    ctx.fillStyle=this.color;
    ctx.fill();
  }

  newPos = function(){//change speed to be updated in next game update
    this.x+=this.speedX;
    this.y+=this.speedY;
  }
}

function clear(){
  context.clearRect(0, 0, this.canvas.width, this.canvas.height);
}

function updateCanvas() {
  clear();
  var work;
  for(var i=0;i<array.length;i++){
    work=array[i];
    work.newPos();
    work.update(context);
    work.radius=work.radius+work.rate;
    if (work.x<=0||work.x>=window.innerWidth){
      array.splice(i,1);
      i--;
    }else if (work.y<=0||work.y>=window.innerHeight){
      array.splice(i,1);
      i--;
    }
  }
}

function generateRandom(min, max) {
  return num = (Math.random() * (max - min)) + min;
}


src/background.ts
let context: CanvasRenderingContext2D;
let array: Component[] = [];
let update: number = null;
let reset: number = null;
let canvas: HTMLCanvasElement;

class Component {
  public radius: number;
  x: number;
  y: number;
  rate: number;
  private speedX: number;
  private speedY: number;

  constructor(radius: number, x: number, y: number) {
    this.radius = radius;
    this.x = x;
    this.y = y;
    this.setSpeed();
    this.rate = Math.random() / 150 + 0.02;
  }

  public setSpeed(): void {
    this.speedX = generateRandom(-3, 3); // represents movements in x direction per game update
    this.speedY = generateRandom(-3, 3);
    if (Math.abs(this.speedX) + Math.abs(this.speedY) < 2) {
      this.setSpeed();
    }
  }

  public update(ctx: CanvasRenderingContext2D): void {
    ctx.beginPath();
    ctx.arc(this.x, this.y, this.radius, 0, 2 * Math.PI);
    ctx.fillStyle = 'white';
    ctx.fill();
  }

  public newPos(): void {
    // change speed to be updated in next game update
    this.x += this.speedX;
    this.y += this.speedY;
  }
}

function startAnim(): void {
  for (let i = 0; i < 5; i++) {
    array.push(new Component(generateRandom(0.5, 4), window.innerWidth / 2, window.innerHeight / 2));
  }
}

window.addEventListener('DOMContentLoaded', (evt) => {
  makeCanvas();
  window.focus();
});

window.onfocus = function (): void {
  if (!reset) {
    reset = window.setInterval(startAnim, 280);
  }
  if (!update) {
    update = window.setInterval(updateCanvas, 20);
  }
};

window.onblur = function (): void {
  if (reset) {
    window.clearInterval(reset);
    reset = null;
  }
  if (update) {
    window.clearInterval(update);
    update = null;
  }
};

window.onresize = function (event: UIEvent): void {
  refreshCanvas();
};

function refreshCanvas(): void {
  canvas.width = window.innerWidth;
  canvas.height = window.innerHeight;
}

function makeCanvas(): CanvasRenderingContext2D {
  startAnim();
  canvas = document.getElementById('canvas') as HTMLCanvasElement;
  refreshCanvas();

  if (canvas.getContext) {
    context = this.canvas.getContext('2d');
    document.body.insertBefore(this.canvas, document.body.childNodes[0]);
  }

  return context;
}

function clear(): void {
  context.clearRect(0, 0, this.canvas.width, this.canvas.height);
}

function updateCanvas(): void {
  clear();
  let work: Component;
  for (let i = 0; i < array.length; i++) {
    work = array[i];
    work.newPos();
    work.update(context);
    work.radius = work.radius + work.rate;
    if (work.x <= 0 || work.x >= window.innerWidth) {
      array.splice(i, 1);
      i--;
    } else if (work.y <= 0 || work.y >= window.innerHeight) {
      array.splice(i, 1);
      i--;
    }
  }
}

function generateRandom(min: number, max: number): number {
  return (Math.random() * (max - min)) + min;
}
